### PR TITLE
[KP-1036] assorted fixes and enhancements

### PR DESCRIPTION
**Pull request by kenskyfish**

### CODE REVIEW

File: helpers.js [hunk 1]
```diff
@@ -1,19 +1,33 @@
 const util = require("util");
 const childProcess = require("child_process");
-const { access } = require("fs/promises");
-const fs = require("fs");
+const fs = require("fs/promises");
 
 const exec = util.promisify(childProcess.exec);
 
 async function assertPathExistence(path) {
```
Comment (xkcm) [inline helpers.js]: This function no longer asserts that the path exists, but it checks if a path exists and returns a boolean, you should rename this function to `doesPathExist` or `pathExists`.

File: helpers.js [hunk 2]
```diff
@@ -1,19 +1,33 @@
...
   try {
-    await access(path, fs.constants.F_OK);
+    await fs.access(path, fs.constants.F_OK);
```
Comment (xkcm) [inline helpers.js]: Module `fs/promises` doesn't have the `constants` member, this probably throws an error `Uncaught TypeError: Cannot read properties of undefined (reading 'F_OK')`

File: helpers.js [hunk 3]
```diff
@@ -1,19 +1,33 @@
...
+    return true;
   } catch {
-    throw new Error(`Path ${path} does not exist`);
+    return false;
+  }
+}
+
+async function getFileContent({ PATH: path }) {
+  if (assertPathExistence(path)) {
```
Comment (xkcm) [inline helpers.js]: Forgot the `await` before calling `assertPathExistence`.

**Review by xkcm — CHANGES_REQUESTED** [review]

File: helpers.js (hunk 2 above)
Comment (kenskyfish) [reply helpers.js]: It seems to have constants and works... https://nodejs.org/dist/latest-v10.x/docs/api/fs.html#fs_fs_constants

File: helpers.js (hunk 1 above)
Comment (kenskyfish) [reply helpers.js]: now it returns the type path - directory, file, or nonexistent.

File: helpers.js (hunk 3 above)
Comment (kenskyfish) [reply helpers.js]: how embarrassing.

**Review by xkcm — APPROVED** [review]

**Pull request merged**